src/main/java/org/isf/medicalinventory/manager/MedicalInventoryManager.java

```diff
@@ -542,66 +617,77 @@ public List<Movement> confirmMedicalInventoryRow(MedicalInventory inventory, Lis
 	 *
 	 * @param inventory the {@link MedicalInventory}
 	 * @param inventoryRowSearchList- The list of {@link MedicalInventory}
-	 * @return List {@link Movement}. It could be {@code empty}.
+	 * @param allMedicals if {@code true}, it will add new {@link MedicalInventoryRow}s if found in the latest stock movements. If {@code false}, only existing
+	 *        rows will be updated.
+	 * @return List of {@link MovementWard}s. It could be {@code empty}.
 	 * @throws OHServiceException
 	 */
 	@Transactional(rollbackFor = OHServiceException.class)
-	public boolean confirmMedicalWardInventoryRow(MedicalInventory inventory, List<MedicalInventoryRow> inventoryRowSearchList) throws OHServiceException {
+	public List<MovementWard> confirmMedicalWardInventoryRow(MedicalInventory inventory, List<MedicalInventoryRow> inventoryRowSearchList, boolean allMedicals)
+		throws OHServiceException {
 		// validate the inventory
-		this.validateMedicalWardInventoryRow(inventory, inventoryRowSearchList);
+		this.validateMedicalWardInventoryRow(inventory, inventoryRowSearchList, allMedicals);
 
 		// get general info
 		Ward selectedWard = wardManager.findWard(inventory.getWard());
 		LocalDateTime now = TimeTools.getNow();
-		String reason = MessageBundle.getMessage("angal.inventory.reason.msg").concat(inventory.getInventoryReference());
+		String reason = inventory.getInventoryReference();
+		List<MovementWard> insertedMovements = new ArrayList<>();
 		for (MedicalInventoryRow medicalInventoryRow : inventoryRowSearchList) {
 			double theoQty = medicalInventoryRow.getTheoreticQty();
 			double realQty = medicalInventoryRow.getRealQty();
 			Double movQuantity = theoQty - realQty;
```

Comment:
why isn't this just a primitive (`double`) instead of a wrapper type?

Reply:
no reason... thanks!